### PR TITLE
clear remaining warnings, and add suppressions

### PR DESCRIPTION
**Pull request by weijjia**
Clear the rest of the warning in #682. 
Suppress the PMD and findbugs if the warning doesn't make sense. 
Clear checkstyle warnings in HttpUrlConnectionFactory, also make corresponding changes in the tests. 
Clear those files that were marked as clean before.

### CODE REVIEW

File: adal/src/main/java/com/microsoft/aad/adal/WebRequestHandler.java [hunk 1]
```diff
@@ -23,16 +23,15 @@
 
 package com.microsoft.aad.adal;
 
+import android.os.Build;
+
+import com.microsoft.aad.adal.AuthenticationConstants.AAD;
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/WebRequestHandler.java]: >com.microsoft.aad.adal.AuthenticationConstants.AAD [](start = 7, length = 50)

OPTIONAL
you can just do import AuthenticationConstants.AAD

File: adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenSilentHandlerTest.java [hunk 2]
```diff
@@ -767,8 +767,8 @@ private void clearCache(final ITokenCacheStore cacheStore) {
     }
 
     private AuthenticationRequest getAuthenticationRequest(final String authority, final String resource,
-                                                           final String clientId, final boolean IsExtendedLifetimeEnabled) {
-        final AuthenticationRequest authRequest = new AuthenticationRequest(authority, resource, clientId, IsExtendedLifetimeEnabled);
+                                                           final String clientId, final boolean isExtendedLifetimeEnabled) {
+        final AuthenticationRequest authRequest = new AuthenticationRequest(authority, resource, clientId, isExtendedLifetimeEnabled);
         return authRequest;
```
Comment (addev-ashish) [inline adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenSilentHandlerTest.java]: aren't you getting a warning about collapsing these 2 lines?

**Comment by addev-ashish** [comment]
:shipit:

File: adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenSilentHandlerTest.java (hunk 2 above)
Comment (weijjia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenSilentHandlerTest.java]: ah? I'm not collapsing lines. Only thing changed is IsExtendedLife** to start with lower case. :)

Comment (addev-ashish) [inline adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenSilentHandlerTest.java]: clarified, Wei will collapse the 2 line

**Pull request merged**